Tests/AudioKitTests/Node Tests/Generator Tests/AudioPlayer2Tests+Realtime.swift

```diff
@@ -0,0 +1,198 @@
+import AudioKit
+import AVFoundation
+import CAudioKit
+import XCTest
+
+// Real time functions, for local testing only.
+// These simulate a user interacting with the player via an UI
+// Real time functions, for local testing only
+
+// Thse are organized like this so they're easy to comment out
+extension AudioPlayer2Tests {
+//    func testPause() {
+//        realtimeTestPause()
+//    }
+//
+//    func testScheduled() {
+//        realtimeScheduleFile()
+//    }
+//
+//    func testFileLooping() {
+//        realtimeLoop(buffered: false, duration: 2)
+//    }
+//
+//    func testBufferLooping() {
+//        realtimeLoop(buffered: true, duration: 1)
+//    }
+//
+//    func testInterupts() {
+//        realtimeInterrupts()
+//    }
+}
+
+extension AudioPlayer2Tests {
+    func realtimeTestPause() {
+        guard let player = createPlayer(duration: 6) else {
+            XCTFail("Failed to create AudioPlayer2")
+            return
+        }
+        let engine = AudioEngine()
+        engine.output = player
+        try? engine.start()
+
+        player.completionHandler = { Log("🏁 Completion Handler") }
+        var duration = player.duration
+
+        Log("▶️")
```

Comment:
I dig your emojis. I started doing that in my own code.

Reply:
🎃